src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -48,58 +48,54 @@ protected override Task RemoveTopicSubscriptionAsync()
     {
         if (_logger.IsEnabled(LogLevel.Trace))
             _logger.LogTrace("RemoveTopicSubscriptionAsync");
-        CloseSubscriberConnection();
-        return Task.CompletedTask;
+
+        return CloseSubscriberConnectionAsync();
     }
 
     protected override async Task EnsureTopicSubscriptionAsync(CancellationToken cancellationToken)
     {
-        if (_subscriberModel != null)
+        if (_subscriberChannel != null)
             return;
 
         await EnsureTopicCreatedAsync(cancellationToken).AnyContext();
 
         using (await _lock.LockAsync().AnyContext())
         {
-            if (_subscriberModel != null)
+            if (_subscriberChannel != null)
                 return;
 
-            _subscriberConnection = CreateConnection();
-            _subscriberModel = _subscriberConnection.CreateModel();
+            _subscriberConnection = await CreateConnectionAsync().AnyContext();
+            _subscriberChannel = await _subscriberConnection.CreateChannelAsync(cancellationToken: cancellationToken).AnyContext();
 
-            // If InitPublisher is called first, then we will never come in this if clause.
-            if (!CreateDelayedExchange(_subscriberModel))
+            // If InitPublisher is called first, then we will never come in this if-clause.
+            if (!await CreateDelayedExchangeAsync(_subscriberChannel).AnyContext())
             {
-                _subscriberModel.Close();
-                _subscriberModel.Abort();
-                _subscriberModel.Dispose();
-
-                _subscriberConnection.Close();
-                _subscriberConnection.Dispose();
+                await _subscriberChannel.DisposeAsync().AnyContext();
+                await _subscriberConnection.DisposeAsync().AnyContext();
```

Comment:
Dispose will automatically look at if it's open and call abort without throwing. I've updated all usages of the code to call this instead.